### PR TITLE
add datasource for policy_set

### PR DESCRIPTION
**Pull request by Uk1288**
## Description

Add datasource for policy set. This PR is in relation to the `tfe_workspace_policy_set` work here https://github.com/hashicorp/terraform-provider-tfe/pull/591

_Remember to:_

- _Update the [Change Log](https://github.com/hashicorp/terraform-provider-tfe/blob/main/README.md#updating-the-changelog)_

- _Update the [Documentation](https://github.com/hashicorp/terraform-provider-tfe/blob/main/README.md#updating-the-documentation)_

## Testing plan
Note: Policy set is a paid plan.
1.  _Create a policy set and then use datasource like:_ 
```
data "tfe_policy_set" "readable_policy_set" {
  name = "my-policy-set"
  organization = "awesome-org"
}
```

## External links

_Include any links here that might be helpful for people reviewing your PR. If there are none, feel free to delete this section._

- [Related PR](https://github.com/hashicorp/terraform-provider-tfe/pull/591)

## Output from acceptance tests

_Please run applicable acceptance tests locally and include the output here. See TESTS.md to learn how to run acceptance tests._

_If you are an external contributor, your contribution(s) will first be reviewed before running them against the project's CI pipeline._

```
$ TESTARGS="-run TestAccTFEWorkspace" make testacc

...
```
Screen Shot 2022-08-18 at 10 56 35 AM
Screen Shot 2022-08-18 at 10 56 46 AM

### CODE REVIEW

File: tfe/data_source_policy_set_test.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	"math/rand"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccTFEPolicySetDataSource_basic(t *testing.T) {
+	skipIfFreeOnly(t)
```
Comment (sebasslash) [inline tfe/data_source_policy_set_test.go:14]: Make sure you rebase once #602 is merged :+1:

File: tfe/data_source_policy_set_test.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+						"data.tfe_policy_set.bar", "vcs_repo.#", "1"),
+				),
+			},
+		},
+	},
+	)
+}
+
+func TestAccTFEPolicySetDataSource_notFound(t *testing.T) {
+	skipIfFreeOnly(t)
```
Comment (sebasslash) [inline tfe/data_source_policy_set_test.go:97]: We can remove this since the API returns a 404 if the user is not found or authorized.

**Review by sebasslash — APPROVED** [review]
Looks good to me :rocket:

File: tfe/data_source_policy_set_test.go (hunk 2 above)
Comment (Uk1288) [reply tfe/data_source_policy_set_test.go:97]: as per discussed, this will be added on next PR.

**Comment by Uk1288** [comment]
> 

I have updated the test plan to indicate that policy sets is a paid plan. I am not sure the comment: `to successfully replicate the behavior without the following: a, b, c` means. Let's discuss.

**Pull request merged**